go/tools/bazel/bazel.go

```diff
@@ -20,43 +20,72 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"sync"
+	"sync/atomic"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel/runfiles"
 )
 
 const TEST_SRCDIR = "TEST_SRCDIR"
 const TEST_TMPDIR = "TEST_TMPDIR"
 const TEST_WORKSPACE = "TEST_WORKSPACE"
 
-var defaultTestWorkspace = ""
+var (
+	defaultTestWorkspace = ""
+
+	// Runfiles resolver state. We can't use sync.Once because of error handling.
```

Comment:
I think you can actually use `sync.Once` here. Declare separate variables for `runfileResolver` and `runfileResolverErr error` and set both inside the initialization function.

Reply:
You can, but of course that means you won't retry again in case of error. At first I thought that would be undesirable for tests, but then I realized it's also kind of undesirable for tests if your success is cached indefinitely, so may as well go both ways. Done.